src/composables/usePool.ts

```diff
@@ -437,6 +437,17 @@ export function usePool(pool: Ref<AnyPool> | Ref<undefined>) {
   const noInitLiquidityPool = computed(
     () => !!pool.value && noInitLiquidity(pool.value)
   );
+  const notAllowedRateProviders = computed(
+    () =>
+      pool.value?.poolType === 'Weighted' &&
```

Comment:
We have a helper function for this:
```suggestion
      isWeighted(pool.value?.poolType) &&
```